src/blocks/watson.rs

```diff
@@ -64,10 +64,25 @@ impl ConfigBlock for Watson {
         shared_config: SharedConfig,
         tx_update_request: Sender<Task>,
     ) -> Result<Self> {
+        let state_path_string: String = block_config
+            .state_path
+            .clone()
+            .into_os_string()
+            .into_string()
+            .unwrap();
```

Comment:
No need to clone here: just use `state_path.to_str().unwrap()`.

P.S. not sure if panicking on non UTF8 path is the right thing to do, perhaps  `state_path.to_string_loosy()`?

Reply:
I saw the `state_path.to_string_loosy()`, and wasn't sure on what to do, that's why I cloned. I am leaning towards panicking.
The idea is that if the user uses a unicode path it's fine, as expected, but when it isn't unicode:
* `state_path.to_str()`: panics, throwing a hopefully helpful message, causing the user to rename some folder if it wants to continue using the block;
* `state_path.to_string_lossy()`: replaces sequences with an arbitrary unicode character, making Watson most certainly not work, and not generating errors at first. Error handling goes to Watson but the error was generated on i3status-rust (string encoding conversion error).

Reply:
> state_path.to_str(): panics, throwing a hopefully helpful message

Well, the message in this particular case will not be that helpful to the user, as `to_str()` returns an `Option`. 

> replaces sequences with an arbitrary unicode character

Fair point, but I would prefer `.block_error(...)?`ing in this case (in general panics should be avoided everywhere).

Reply:
I am experimenting with
```
let state_path_string = match block_config.state_path.to_str() {
    Some(str) => str,
    None => {
        return Err(BlockError(
            "watson".to_string(),
            format!(
                        "Error converting state_path to unicode string: '{}'. Can you try renaming one of the folders?",
                        block_config.state_path.to_string_lossy()
                    ),
        ))
    }
};
```
Or that isn't helpful enough to justify it? That we still can detect if the string is not properly decoded.

Reply:
Also I don't really have much of a clue on how to test this on my machine... I've never dealt with unusual encodings, at least not as far as I know.

Reply:
> Or that isn't helpful enough to justify it

You mean verbose error message? IDK. But this is definitely better than panicking.

Reply:
> I don't really have much of a clue on how to test this on my machine

Here you go: `touch $(printf 'non_utf_\xaa')` :laughing:

Reply:
This is probably a non-problem. Trying to load while pointing the non-UTF named file creates 
```
{"version": 1, "click_events": true}
[
[{"full_text":" Internal error in context 'util': failed to read file. Cause: stream did not contain valid UTF-8 ","color":"#DC322FFF","background":"#000000FF","name":"0","instance":"0","separator":false,"separator_block_width":0,"markup":"pango"}],


Internal error in context 'util': failed to read file. Cause: stream did not contain valid UTF-8
```
So I suppose we can still use the match I made just for being safe?

Reply:
It creates an error before we get to block-level logic. 
```
{"version": 1, "click_events": true}
[
thread '<unnamed>' panicked at 'failed to watch watson state file: Os { code: 2, kind: NotFound, message: "No such file or directory" }', src/blocks/watson.rs:121:18
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
[{"full_text":" Error in block 'watson': unable to open state file ","color":"#DC322FFF","background":"#000000FF","name":"0","instance":"0","separator":false,"separator_block_width":0,"markup":"pango"}],


Error in block 'watson': unable to open state file
```
I saw some of my i3 config there and tought maybe I screwed up, but no, it works fine when I delete just the non-utf char, watson block then says there is no such file (as expected). That's why I deleted last comment.